### ISSUE TITLE
Syncoid remote-to-remote decompresses and recompresses on the middle box

### ISSUE THREAD

**Issue by fallingcats** [jimsalterjrs/sanoid#896]
When using syncoid to transfer snapshots from one remote location to another, syncoid seems to decompress and recompress the streams on the box that's initiated the transfer.

I'm using this command to run syncoid, but I suspect it would happen in any configuration:
```
/usr/sbin/syncoid --create-bookmark --no-sync-snap --compress=zstd-slow --source-bwlimit=1M --mbuffer 1G -r $REMOTE_SOURCE $REMOTE_TARGET
```

**Comment by jimsalterjrs** [comment]
You're correct, this will definitely happen. I'm happy to review a PR to fix it, but I don't personally have the bandwidth to chase this one right now--I included the ability to do a double-remote sync from the very first release, but you're the first person I've ever heard of actually *using* that feature in the wild!
